app/src/main/java/com/twilio/video/app/sdk/VideoTrackViewState.kt

```diff
@@ -0,0 +1,8 @@
+package com.twilio.video.app.sdk
+
+import com.twilio.video.VideoTrack
+
+data class VideoTrackViewState(
```

Comment:
We can discuss this offline, but I am starting to wonder what is the difference between using a view state object and an Android arch `ViewModel`? Why not represent each container of data that will be rendered to the UI as a `ViewModel` instead?

Reply:
Good question! So with the MVVM pattern, the ViewModel's responsibility is to essentially provide observable UI data to the view (i.e. Fragment or Activity), and cache that data so it can survive Activity configuration changes. So the view only needs a single ViewModel to take care of serving the UI data to render. The ViewState object is a simple data class that just encapsulates the UI data. This is really handy as it allows the ViewModel to provide a single stream of data with the ViewState type, opposed to having multiple LiveData streams represent each piece of data. Also, if you had additional ViewModels, then you would have to observe additional streams of data from the view which would add more complexity. I hope that helps! I'm happy to hop on a call and talk about this more if needed as I really enjoy discussing Android architecture 😃